artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/proton/ProtonServerSenderContext.java

```diff
@@ -319,7 +319,7 @@ public void initialise() throws Exception {
          if (CompositeAddress.isFullyQualified(source.getAddress())) {
             addressToUse = SimpleString.toSimpleString(CompositeAddress.extractAddressName(source.getAddress()));
             queueNameToUse = SimpleString.toSimpleString(CompositeAddress.extractQueueName(source.getAddress()));
-         } else {
+         } else if (source.getAddress() != null) {
```

Comment:
Test?

Reply:
An alternate is to use the static method on SimpleString to convert String to SimpleString which has the null check.